### PR TITLE
DEV-7836 ARP filter - recipient section

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Filters the Covid-19 Profile Award Spending by Recipient section data based on the `publicLaw` URL query param.

**Technical details:**

- Store a new list of Covid 19 Profile DEF codes in Redux (since existing `defCodes` are used elsewhere on the site too)
  - this is a flat list of just the string for the code, e.g. `(["L", "M", "N", "O", "P", "U", "V"])`
- Update this list when the `publicLaw` query param changes (in Covid19Container.jsx)
  - `all`: every DEFC returned by useDefCodes hook
  - `american-rescue-plan`: use the defcByPublicLaw mapping
- Update the places we use existing `defCodes` to create API params to use this new list instead

Also changed file names for clarity
- SpendingByRecipient --> SpendingByRecipientContainer (moved this to `/container` directory from `/component`)
- SpendingByRecipientContainer --> RecipientTableContainer
- RecipientContainer --> RecipientSection (moved this to `/component` directory from `/container`)

**JIRA Ticket:**
[DEV-7836](https://federal-spending-transparency.atlassian.net/browse/DEV-7836)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)

Reviewer(s):
- [x] Code review complete

### CODE REVIEW

File: src/js/containers/covid19/recipient/RecipientTableContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                    rowData.awardCount
+                ]]);
+            setParsedRows(rowsWithUnlinked);
+        }
+        else {
+            setParsedRows(rows);
+        }
+    };
+
+    const fetchSpendingByRecipientCallback = useCallback(() => {
```
Comment (brett-varney) [inline src/js/containers/covid19/recipient/RecipientTableContainer.jsx:219]: @ebdabbs I see this warning in ESlint:

React Hook useCallback does nothing when called with only one argument. Did you forget to pass an array of dependencies?

**Review by brett-varney — CHANGES_REQUESTED** [review]
question, not necessarily change request

**Comment by brett-varney** [comment]
After a couple of test runs, it started calling APIs with no def codes (empty array). I'll poke around to see if I can see why.

**Review by brett-varney — CHANGES_REQUESTED** [review]
Seems to work, but I'm concerned about the call to /recipients/count with no defcodes. It isn't happening before this update.

File: src/js/containers/covid19/recipient/RecipientTableContainer.jsx (hunk 1 above)
Comment (brett-varney) [reply src/js/containers/covid19/recipient/RecipientTableContainer.jsx:219]: But now I see that this was there before.

Comment (ebdabbs) [reply src/js/containers/covid19/recipient/RecipientTableContainer.jsx]: Removing `useCallback` since I don't think it's necessary here.

**Comment by ebdabbs** [comment]
Added a check for the empty DEFC array to `SpendingByRecipientContainer`

**Review by brett-varney — APPROVED** [review]

**Pull request merged**